plugins/MauticCrmBundle/Integration/SalesforceIntegration.php

```diff
@@ -1294,6 +1295,8 @@ public function pushLeads($params = [])
                 }
             }
 
+            $this->getLeadDoNotContactByDate('email', $checkEmailsInSF);
```

Comment:
We should only be checking this if the do not contact field is mapped otherwise we are making unnecessary queries on both sides.